crates/cxx-qt-build/src/lib.rs

```diff
@@ -241,11 +237,17 @@ fn generate_cxxqt_cpp_files(
         let path = format!("{manifest_dir}/{}", rs_path.display());
         println!("cargo:rerun-if-changed={path}");
 
-        let generated_code = GeneratedCpp::new(&path)?;
+        let generated_code = match GeneratedCpp::new(&path) {
+            Ok(v) => v,
+            Err(diagnostic) => {
+                diagnostic.report();
+                std::process::exit(1);
+            }
+        };
```

Comment:
Moving this error handling here avoids some duplicate code because this function is now called twice in CxxQtBuilder::build. There's nothing else useful to do with the error and the process exits here, so there's no point propagating the error upward.

Reply:
Sure, doesn't look like this would break anything in @LeonMatthesKDAB 's diagnostics battle :-)